### PR TITLE
Add guest users

### PR DESCRIPTION
**Pull request by andela-osule**
@andela-ooshodi Please review. Thanks

### CODE REVIEW

File: waitress/app/models.py [hunk 1]
```diff
@@ -21,16 +21,23 @@ def create_user(self, *args, **kwargs):
...
     A class that represents a `SlackUser` account
     """
     id = models.AutoField(unique=True, primary_key=True)
-    slack_id = models.CharField(unique=True, max_length=20)
+    slack_id = models.CharField(unique=True, max_length=20, blank=True)
     firstname = models.CharField(max_length=20,)
     lastname = models.CharField(max_length=20,)
-    email = models.CharField(max_length=60,)
+    email = models.CharField(max_length=60, blank=True)
+    user_type = models.CharField(
```
Comment (andela-ashuaib) [inline waitress/app/models.py]: Whats the use of user_type at the long run?

File: waitress/app/utils.py [hunk 2]
```diff
@@ -14,6 +14,26 @@ class UserRepository(object):
     A wrapper class for methods that update the list of users.
     """
 
+    @staticmethod
+    def add(user_type, username):
```
Comment (andela-ashuaib) [inline waitress/app/utils.py]: Does this takes care of just gets or as well, or guests along with new fellows?

File: waitress/app/models.py (hunk 1 above)
Comment (andela-osule) [inline waitress/app/models.py]: Slack doesn't have us know if a user is a fellow, trainer or staff.
We can only allow for two type at this time: `employee` and `guest`

File: waitress/app/utils.py (hunk 2 above)
Comment (andela-osule) [inline waitress/app/utils.py]: This was written to accommodate creation of any type of user. For the moment, what we need is just a guest.

File: waitress/app/utils.py [hunk 3]
```diff
@@ -114,8 +134,8 @@ def normalize(info):
         except Exception as e:
             return "Users couldn't be updated successfully - %s" % e.message
 
-    @classmethod
-    def get_real_users(cls, difference, not_user):
+    @staticmethod
+    def get_real_users(difference, not_user):
```
Comment (andela-ashuaib) [inline waitress/app/utils.py]: Where are this deocrators defined?

Comment (andela-ashuaib) [inline waitress/app/utils.py]: My bad. I mean what's this function use?

File: waitress/app/viewsets.py [hunk 4]
```diff
@@ -65,6 +65,19 @@ def trim_users(self, request):
 
         return Response(content, status=status_code.HTTP_200_OK)
 
+    @list_route(methods=['post'], url_path='add-guest')
```
Comment (andela-ashuaib) [inline waitress/app/viewsets.py]: Where are this decorators defined and what's the use?

File: waitress/app/utils.py (hunk 3 above)
Comment (andela-osule) [inline waitress/app/utils.py]: The `get_real_users` function receives the difference between the user list in the database and that from Slack. The `not user` argument is a list of bots, deleted users...
If a user was deleted at any time, this function helps remove them from the db, thus keeping the db updated.

File: waitress/app/viewsets.py (hunk 4 above)
Comment (andela-osule) [inline waitress/app/viewsets.py]: @list_route is a `django rest framework` decorator for defining a route that encapsulates a list view.

File: waitress/app/utils.py (hunk 2 above)
Comment (andela-osule) [inline waitress/app/utils.py]: `/users/update-users/` endpoint handles addition of new fellows so long as they are on Slack and in the fellows lagos channel

**Pull request merged**